diff --git a/classes/DataWarehouse/Access/Usage.php b/classes/DataWarehouse/Access/Usage.php
index df843d73..66566b6e 100644
--- a/classes/DataWarehouse/Access/Usage.php
+++ b/classes/DataWarehouse/Access/Usage.php
@@ -1266,6 +1266,12 @@ class Usage extends Common
             case 'resource':
                 $query = "SELECT id AS value FROM modw.resourcefact WHERE code = :value";
                 break;
+            case 'project':
+                /*
+                * THE FOLLOWING HAS BEEN CHANGED BY THE FEDERATION MODULE
+                */
+                $query = "SELECT primaryid AS value FROM modw_cloud.account WHERE display LIKE :value";
+                break;
         }

         $filterValueIsString = preg_match('/^[\'"].*[\'"]$/', $usageFilterValue) === 1;
@@ -1278,7 +1284,10 @@ class Usage extends Common
             $db = DB::factory('database');

             $stmt = $db->prepare($query);
-            $stmt->execute(array(':value' => $value));
+            /*
+             * THE FOLLOWING HAS BEEN CHANGED BY THE FEDERATION MODULE
+             */
+            $stmt->execute(array(':value' => $value . '%'));

             $rows = $stmt->fetchAll(PDO::FETCH_COLUMN, 0);

diff --git a/classes/OpenXdmod/DataWarehouseInitializer.php b/classes/OpenXdmod/DataWarehouseInitializer.php
index e282f658..24780ff1 100644
--- a/classes/OpenXdmod/DataWarehouseInitializer.php
+++ b/classes/OpenXdmod/DataWarehouseInitializer.php
@@ -515,29 +515,6 @@ class DataWarehouseInitializer
      */
     public function getEnabledRealms()
     {
-        if ($this->enabledRealms !== null) {
-            return $this->enabledRealms;
-        }
-
-        $resources = XdmodConfiguration::assocArrayFactory('resources.json', CONFIG_DIR, null, array('force_array_return' => true));
-        $resourceTypes = XdmodConfiguration::assocArrayFactory('resource_types.json', CONFIG_DIR)['resource_types'];
-
-        $currentResourceTypes = array();
-        foreach($resources as $resource) {
-            if (isset($resource['resource_type'])) {
-                $currentResourceTypes[] = $resource['resource_type'];
-            }
-        }
-        $currentResourceTypes = array_unique($currentResourceTypes);
-
-        $realms = array();
-        foreach($currentResourceTypes as $currentResourceType) {
-            if (isset($resourceTypes[$currentResourceType])) {
-                $realms = array_merge($realms, $resourceTypes[$currentResourceType]['realms']);
-            }
-        }
-        $this->enabledRealms = array_unique($realms);
-
-        return $this->enabledRealms;
+        return \Models\Services\Realms::getEnabledRealms();
     }
 }
